azurelinuxagent/ga/collect_telemetry_events.py

```diff
@@ -0,0 +1,585 @@
+                        logger.warn(msg)
+                        add_log_event(level=logger.LogLevel.WARNING, message=msg, forced=True)
+                        break
+                except ServiceStoppedError:
+                    # Not logging here as already logged once, re-raising
+                    # Since we already started processing this file, deleting it as we could've already sent some events out
+                    # This is a trade-off between data replication vs data loss.
+                    raise
+                except Exception as error:
+                    msg = "Failed to process event file {0}: {1}, {2}".format(event_file, ustr(error),
+                                                                              traceback.format_exc())
+                    logger.warn(msg)
+                    add_log_event(level=logger.LogLevel.WARNING, message=msg, forced=True)
+                finally:
+                    # Todo: We should delete files after ensuring that we sent the data to Wireserver successfully
+                    # from our end rather than deleting first and sending later. This is to ensure the data reliability
+                    # of the agent telemetry pipeline.
+                    os.remove(event_file_path)
+
+        finally:
+            if dropped_events_with_error_count:
+                msg = "Dropped events for Extension: {0}; Details:\n\t{1}".format(handler_name, '\n\t'.join(
+                    ["Reason: {0}; Dropped Count: {1}".format(k, v) for k, v in dropped_events_with_error_count.items()]))
+                logger.warn(msg)
+                add_log_event(level=logger.LogLevel.WARNING, message=msg, forced=True)
+
+            if captured_extension_events_count > 0:
+                logger.info("Collected {0} events for extension: {1}".format(captured_extension_events_count, handler_name))
+
+    @staticmethod
+    def _ensure_all_events_directories_empty(extension_events_directories):
+        if not extension_events_directories:
+            return
+
+        for extension_handler_with_event_dir in extension_events_directories:
+            event_dir_path = extension_handler_with_event_dir[1]
+            if not os.path.exists(event_dir_path):
+                return
+
+            err = None
+            # Delete any residue files in the events directory
+            for residue_file in os.listdir(event_dir_path):
+                try:
+                    os.remove(os.path.join(event_dir_path, residue_file))
+                except Exception as error:
+                    # Only log the first error once per handler per run if unable to clean off residue files
+                    err = ustr(error) if err is None else err
+
+                if err is not None:
+                    logger.error("Failed to completely clear the {0} directory. Exception: {1}", event_dir_path, err)
+
+    def _get_captured_events_count(self, handler_name, event_file_path, captured_events_count,
```

Comment:
This method does much more than retrieve the count of captured events. I was finding it hard to figure out where we actually enqueue events, and the name does not help here. Consider renaming or breaking the function into smaller pieces of logic.

Reply:
I think I was being over-smart with the name. I was going for `captured_events` and trying to imply that it already enqueued lol. But I get your point, renamed :)